test/sqllogictest/in_list.slt

```diff
@@ -23,3 +26,79 @@ SELECT b FROM t1 WHERE a IN (3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,2
 l3
 xxx
 yyy
+
+# IN list translated to ReadExistingIndex
+
+query T multiline
+EXPLAIN SELECT * FROM t1 WHERE a IN (1,2)
+----
+%0 =
+| ReadExistingIndex materialize.public.t1_primary_idx
+| Filter ((#0 = 1) OR (#0 = 2))
+
+EOF
+
+# PredicateIndex join executed in fast path
+
+query T multiline
+EXPLAIN SELECT a, b FROM t1
+WHERE a = 2 AND b = 'l2'
+----
+%0 =
+| ReadExistingIndex materialize.public.t1_primary_idx
+| | Lookup value (2, "l2")
+| Filter (#0 = 2), (#1 = "l2")
+| Project (#0, #1)
+
+EOF
+
+query IT rowsort
+SELECT a, b FROM t1
+WHERE a = 2 AND b = 'l2'
+----
+2  l2
```

Comment:
Good point. But the problem is that decorrelation kicks in before constant folding, and after decorrelation the constant folding cannot always do its job, for example:

```
materialize=> EXPLAIN SELECT * FROM t1 WHERE f1 IN (SELECT 1 UNION ALL SELECT 3);
          Optimized Plan          
----------------------------------
 %0 = Let l0 =                   +
 | Get materialize.public.t1 (u1)+
 | Project (#0)                  +
 | Distinct group=(#0)           +
                                 +
 %1 =                            +
 | Get materialize.public.t1 (u1)+
 | ArrangeBy (#0)                +
                                 +
 %2 =                            +
 | Get %0 (l0)                   +
 | Filter (#0 = 1)               +
                                 +
 %3 =                            +
 | Get %0 (l0)                   +
 | Filter (#0 = 3)               +
                                 +
 %4 =                            +
 | Union %2 %3                   +
 | Distinct group=(#0)           +
 | ArrangeBy (#0)                +
                                 +
 %5 =                            +
 | Join %1 %4 (= #0 #2)          +
 | | implementation = DeltaQuery +
 | |   delta %1 %4.(#0)          +
 | |   delta %4 %1.(#0)          +
 | Project (#0, #1)              +
```
I'm not sure if we want to open a separate issue. The query seems quite artificial, so maybe it's not so important.

Reply:
I was thinking more of 

```
EXPLAIN SELECT * FROM t1 WHERE f1 IN (f(...), g(...), h(...));
```
Where `f`, `g`, `h` are some expressions that depend on the current lexical scope.

Reply:
By "depend on the current lexical scope", you mean e.g. involving columns from `t1`? So this would be a negative test, because (some of) the things in the IN list are not literals? I'm planning to have such tests in the next PR.

Reply:
This should be positive.

```sql
EXPLAIN SELECT a, b, c FROM t1 WHERE d IN (a + b,  a + c, 1 + 5) WHERE a + b = 5 AND a + c = 4;
```

If it isn't, just mark it as "currently not supported" in a test comment so we can enable it later.

Reply:
Ok, I'll add this in the next PR.